cordova-lib/src/cordova/create.js

```diff
@@ -252,133 +249,116 @@ function create(dir, optionalId, optionalName, cfg, fetchOpt) {
-        Files, and directories not copied include: www, mergers,platforms,
-        plugins, hooks, and config.xml. A template directory, and platform
-        directory must be passed.
-
-        templateDir - Template directory
-        projectDir - Project directory
-         */
-        function copyTemplateFiles(templateDir, projectDir) {
-            var templateFiles;		// Current file
-
-            templateFiles = fs.readdirSync(templateDir);
-
-            // Remove directories, and files that are automatically copied
-            templateFiles = templateFiles.filter(
-                function (value) {
-                    return !(value === 'www' || value === 'mergers' ||
-                    value === 'config.xml' || value === 'hooks');
-                }
-            );
-
-            // Copy each template file
-            for (var i = 0; i < templateFiles.length; i++)
-                shell.cp('-R', path.resolve(templateDir, templateFiles[i]), projectDir);
-        }
-
         try {
-            copyOrLink(paths.www, path.join(dir, 'www'), true);
-            copyOrLink(paths.merges, path.join(dir, 'merges'), true);
-            copyOrLink(paths.hooks, path.join(dir, 'hooks'),
-                paths.hooksLinkable);
-
+            // Copy files from template to project
             if (cfg.lib.www.template)
-                copyTemplateFiles(import_from_path, dir);
+                copyTemplateFiles(import_from_path, dir, isSubDir);
 
-            if (paths.configXml) {
-                if (tryToLink && paths.configXmlLinkable) {
-                    fs.symlinkSync(paths.configXml, path.join(dir, 'config.xml'));
-                } else {
-                    shell.cp(paths.configXml, path.join(dir, 'config.xml'));
-                }
+            // If following were not copied from template, copy from stock app hello world
+            ifNotCopied(paths.www, path.join(dir, 'www'));
+            ifNotCopied(paths.hooks, path.join(dir, 'hooks'));
+            if (!fs.existsSync(path.join(dir, 'config.xml')) && paths.configXml) {
+                shell.cp(paths.configXml, path.join(dir, 'config.xml'));
             }
         } catch (e) {
             if (!dirAlreadyExisted) {
                 shell.rm('-rf', dir);
             }
-            if (process.platform.slice(0, 3) == 'win' && e.code == 'EPERM')  {
-                throw new CordovaError('Symlinks on Windows require Administrator privileges');
-            }
             throw e;
         }
+
+        // Update package.json name and version fields.
+        if (fs.existsSync(path.join(dir, 'package.json'))) {
+            var pkgjson = require(path.resolve(dir, 'package.json'));
+            if (cfg.name) {
+                pkgjson.name = cfg.name.toLowerCase();
+            }
+            pkgjson.version = '1.0.0';
+            fs.writeFile(path.join(dir, 'package.json'), JSON.stringify(pkgjson), function (err) {
```

Comment:
Lets maybe do the write operation as Sync and lets add spaces to the new package.json file. See https://github.com/apache/cordova-lib/blob/9f630271d9c02824c0031ea4cdc0e4955b53d9e2/cordova-lib/src/plugman/registry/manifest.js#L101